pkg/operator/reports.go

```diff
@@ -844,21 +874,31 @@ func isReportNotUsedAsInput(logger log.FieldLogger, report *metering.Report, op
 		if does, depQuery := op.reportQueriesDependOnThisReport(logger, report, reportQueries); does {
 			logger.Infof("ReportQuery, %s exists that uses report %s as input, "+
 				"will not delete though retention period has expired", depQuery, report.Name)
-			return false
+
+			return false, nil
 		}
 	}
-	return true
+	return true, nil
 }
 
+// isReportExpired is a helper function that is responsible for inspecting
+// the @report custom resource and checking whether the user has provided
+// a retention period. In the case that field has been populated, check
+// if we're currently processing an "expired" report and return true,
+// else return false. We define "expired" here using the following calculation:
+// time.Now() > report.Metadata.CreationTimestamp + report.Spec.Expiration.
 func isReportExpired(logger log.FieldLogger, report *metering.Report, currentTime time.Time) bool {
+	// TODO: should this be contingent on the Report object reporting
```

Comment:
Good point. I guess this implementation is probably too literal, the user set an expiration, but it's going to happen before the Report runs, probably not the user's intention.